src/python/WMCore/Services/WMStats/WMStatsWriter.py

```diff
@@ -208,3 +208,37 @@ def replaceRequestTransitionFromReqMgr(self, docs):
             self.couchDB.updateDocument(requestName, self.couchapp,
                         'generalFields',
                         fields={'general_fields': JSONEncoder().encode(requestTransition)})
+    
+    def getArchivedRequests(self):
+        """
+        get list of archived workflow in wmstats db.
+        """
+        
+        options = {"group_level": 1, "reduce": True}
+        
+        results = self.couchDB.loadView(self.couchapp, "allWorkflows", options = options)['rows']
+        requestNames = [x['key'] for x in results]
+        
+        workflowDict = self.reqDB.getStatusAndTypeByRequest(requestNames)
+        archivedRequests = []
+        for request, value in workflowDict.items():
+            if value[0].endswith("-archived"):
+                archivedRequests.append(request)
+        
+        return archivedRequests
+    
+    def deleteDocsByWorkflow(self, requestName):
+        """
+        delete all wmstats docs for a given requestName
+        """
+        view = "allWorkflows"
+        options = {"key": requestName, "reduce": False}
+        docs = self.couchDB.loadView(self.couchapp, view, options = options)['rows']
+   
+        for j in docs:
+            doc = {}
+            doc["_id"]  = j['value']['id']
+            doc["_rev"] = j['value']['rev']
```

Comment:
actually key is set to id and rev.
https://github.com/dmwm/WMCore/pull/6412/files#diff-78a3679504ec44c511e66b686ee0ec1aR4